### PR TITLE
Improve execution of OSU benchmarks

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
1. Execute OSU benchmarks on clusters with 32 compute nodes and multi-threading disabled.
1. Fix scripts to submit OSU benchmarks:
    1. Verbose debugging
    2. Print env
    3. Force slurm as bootstrap mechanism for mpirun
    4. Remove round robin scheduling from mpirun submission

Relevant notes:
* Intel MPI is shipped with an internal Libfabric (1.10.1-impi for Intel MPI 2019, and 1.13.0-impi for Intel MPI 2021).
* Libfabric >1.12 fixed an issue with  and this fix causes worst performances, for this reason we decided to use 3.1.1 results as baseline.

### References
* FI_DELIVERY_COMPLETE fix: https://github.com/ofiwg/libfabric/pull/6092

### Tests
1. Launched locally targeting my personal account.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/efa/test_efa.py [hunk 1]
```diff
@@ -146,7 +145,9 @@ def _test_osu_benchmarks_pt2pt(
             slots_per_instance,
             test_datadir,
         )
-        failures = _check_osu_benchmarks_results(test_datadir, instance, mpi_version, benchmark_name, output)
+        failures = _check_osu_benchmarks_results(
+            test_datadir, instance, num_of_instances, mpi_version, benchmark_name, output
```
Comment (enrico-usai) [inline tests/integration-tests/tests/efa/test_efa.py]: num_of_instances no longer required

File: tests/integration-tests/tests/efa/test_efa.py [hunk 2]
```diff
@@ -181,8 +179,11 @@ def _test_osu_benchmarks_collective(
             num_of_instances,
             slots_per_instance,
             test_datadir,
+            timeout=24,
+        )
+        failures = _check_osu_benchmarks_results(
+            test_datadir, instance, num_of_instances, mpi_version, benchmark_name, output
```
Comment (enrico-usai) [inline tests/integration-tests/tests/efa/test_efa.py]: same here

**Review by enrico-usai — COMMENTED** [review]
TODO: add results for 32 nodes and disabled hyperthreading

File: tests/integration-tests/tests/efa/test_efa.py (hunk 1 above)
Comment (gmarciani) [reply tests/integration-tests/tests/efa/test_efa.py]: Thanks. Fixed in next rev

File: tests/integration-tests/tests/efa/test_efa.py (hunk 2 above)
Comment (gmarciani) [reply tests/integration-tests/tests/efa/test_efa.py]: Thanks. Fixed in next rev

File: tests/integration-tests/tests/common/data/osu/osu_collective_submit_intelmpi.sh [hunk 3]
```diff
@@ -6,9 +6,12 @@ OSU_BENCHMARK_VERSION={{ osu_benchmark_version }}
...
 
 module load intelmpi
-export I_MPI_DEBUG=1
+export I_MPI_DEBUG=10
+
+env
 
 # Run collective benchmark. The collective operations are close to what a real application looks like.
 # NOTE: The test is sized for 4 compute nodes.
 # -np total number of processes to run (all CPUs * 4 nodes)
```
Comment (enrico-usai) [inline tests/integration-tests/tests/common/data/osu/osu_collective_submit_intelmpi.sh]: fix number of nodes

File: tests/integration-tests/tests/efa/test_efa/test_efa/osu_benchmarks/results/c5n.18xlarge/intelmpi/osu_latency [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
-32768                  36.22
-65536                  43.05
-131072                105.29
-262144                155.42
-524288                288.50
-1048576               526.56
-2097152              1029.62
-4194304              1976.25
\ No newline at end of file
+0                       0.66
```
Comment (enrico-usai) [inline tests/integration-tests/tests/efa/test_efa/test_efa/osu_benchmarks/results/c5n.18xlarge/intelmpi/osu_latency]: to be updated

**Review by enrico-usai — DISMISSED** [review]

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**